src/filters/detector.cpp

```diff
@@ -6,10 +6,14 @@ namespace mesos {
 namespace serenity {
 
 Try<Nothing> DetectorFilter::consume(const ResourceUsage& in) {
+  return this->_detect(DividedResourceUsage(in));
+}
+
+Try<Nothing> DetectorFilter::_detect(const DividedResourceUsage& usage) {
```

Comment:
I have very mixed feelings about adding DividedResourceUsage class. I usually have mixed feelings about adding new class, and here your could really use 
```
Try<Nothing> DetectorFilter::_detect(const ResourceUsage& prExecutors, const ResourceUsage& beExecutors) {
```
With small refactor of ```filterPrExecutors``` function (which I have proposed) we should have nice and concise codebase.